compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -228,15 +208,34 @@ struct ResultCache {
 }
 
 impl<'brillig> Context<'brillig> {
-    fn new(use_constraint_info: bool, brillig_info: Option<BrilligInfo<'brillig>>) -> Self {
+    fn new(
+        use_constraint_info: bool,
+        brillig_functions: Option<&'brillig BTreeMap<FunctionId, Function>>,
+    ) -> Self {
+        let interpreter = brillig_functions.and_then(|brillig_functions| {
+            if brillig_functions.is_empty() {
+                None
+            } else {
+                let mut interpreter = Interpreter::new_from_functions(
+                    brillig_functions,
+                    InterpreterOptions { no_foreign_calls: true, ..Default::default() },
+                    std::io::empty(),
+                );
+                // Interpret globals once so that we do not have to repeat this computation on every Brillig call.
```

Comment:
I experimented with interpreting globals only once across all functions. However, the interpreter currently requires a mutable reference so we cannot simply have it live on a `Context` with just lifetimes, we would most likely need a RefCell. As the compilation benchmarks are fine with the interpreter as it is now, I think this is ok.

If we still want to avoid this repeat computation (which may be worth it for cases with large globals) we can also thread the interpreter as a mutable reference argument through our constant folding methods. I have done that in [this PR](https://github.com/noir-lang/noir/pull/9658).